compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/types/AbstractTypeDescriptor.java

```diff
@@ -56,16 +56,21 @@ public ModuleID moduleID() {
     public abstract String signature();
 
     @Override
-    public List<MethodSymbol> builtinMethods() {
-        return new ArrayList<>();
+    public List<LangLibMethod> langLibMethods() {
+        return langLibMethods;
     }
 
     /**
      * Get the BType.
-     * 
+     *
      * @return {@link BType} associated with the type desc
      */
     protected BType getBType() {
         return bType;
     }
+
+    // Setters. Only to be used by the type builder
+    void setLangLibMethods(List<LangLibMethod> langLibMethods) {
+        this.langLibMethods = langLibMethods;
```

Comment:
Can't we make the langlib populate functionality an on demand operation?

Reply:
As discussed in the review the actual building of the associated lang lib methods is done in an on-demand manner.